weatherman/weatherreport.py

```diff
@@ -0,0 +1,64 @@
+"""
+For reading data and generating weather report
+"""
+import os
+
+from utils import replace_string_in_list
+from weatheryearlysummary import WeatherYearlySummary
+
+
+class WeatherReport:
+
+    def __init__(self, path_to_dir):
+        self.__path_to_dir = path_to_dir
+        self.weather_data = dict()
+
+    def __should_ignore_reading(self, data):
+        for i in data:
+            if not i:
+                return True
+        else:
+            return False
+
+    def __parse_line_data(self, header, line):
+        return [line[header.index(attributes)] for attributes in WeatherYearlySummary.KEYS_WEATHER_ATTRIB]
+
+    def __get_file_summary(self, name_of_file):
+        summary = None
+        with open(name_of_file) as weather_file:
+            weather_file.readline()
+            header = [item.strip() for item in weather_file.readline().split(",")]
+            replace_string_in_list(header, WeatherYearlySummary.KEY_DATE_ALTER, WeatherYearlySummary.KEY_DATE)
+
+            for line in weather_file.readlines()[:-1]:
+                temp_data = self.__parse_line_data(header, line.split(","))
+                if not self.__should_ignore_reading(temp_data):
+                    temp_summary = WeatherYearlySummary(temp_data)
+                    if not summary:
+                        summary = temp_summary
+                    else:
+                        summary.update_attributes_summary(temp_summary)
```

Comment:
We are doing this twice. Once here and once in __update_year_summary. We can avoid this and only do it at one place.